### PR TITLE
a user can also login with a Django legacy password

### PR DESCRIPTION
**Pull request by vinyll**
Solves #192

### CODE REVIEW

File: cafebabel/users/forms.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        Copy-pasted from flask_security and use the altered version of
+        `verify_and_update_password`.
+        """
+        if not super(LoginForm, self).validate():
+            return False
+
+        self.user = _datastore.get_user(self.email.data)
+
+        if self.user is None:
+            self.email.errors.append(get_message('USER_DOES_NOT_EXIST')[0])
```
Comment (davidbgk) [inline cafebabel/users/forms.py:36]: How will we deal with translations and `get_message`?

File: cafebabel/users/forms.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    if (use_double_hash(user.password)
+            and not user.password.startswith('pbkdf2_sha256')):
+        verified = _pwd_context.verify(get_hmac(password), user.password)
+    else:
+        # Try with original password.
+        verified = _pwd_context.verify(password, user.password)
+    if verified and _pwd_context.needs_update(user.password):
+        user.password = hash_password(password)
+        _datastore.put(user)
```
Comment (davidbgk) [inline cafebabel/users/forms.py:20]: I'm (happily) surprised that you can access the datastore directly from flask-security, I thought you would have to retrieve it from the app.

**Review by davidbgk — APPROVED** [review]
Really amazed by what you did mate! 😎

File: cafebabel/users/forms.py (hunk 2 above)
Comment (vinyll) [reply cafebabel/users/forms.py:20]: This is at least how flask-security deals with it

File: cafebabel/users/forms.py (hunk 1 above)
Comment (vinyll) [reply cafebabel/users/forms.py:36]: I haven't taken care of that part and I'm not sure yet.
Dealing with translations for here will be the same way we'll deal with them for flask-security and probably other 3rd party libs

**Pull request merged**